libtenzir/include/tenzir/split_at_regex.hpp

```diff
@@ -0,0 +1,75 @@
+//    _   _____   __________
+//   | | / / _ | / __/_  __/     Visibility
+//   | |/ / __ |_\ \  / /          Across
+//   |___/_/ |_/___/ /_/       Space and Time
+//
+// SPDX-FileCopyrightText: (c) 2025 The Tenzir Contributors
+// SPDX-License-Identifier: BSD-3-Clause
+
+#pragma once
+
+#include "tenzir/chunk.hpp"
+#include "tenzir/generator.hpp"
+
+#include <boost/regex.hpp>
+
+#include <optional>
+#include <string_view>
+
+namespace tenzir {
+
+inline auto split_at_regex(std::string_view separator) {
```

Comment:
Nit: Can you move this into a separate .cpp file?

Reply:
The function currently returns a lambda, so not easily. As of right now the header is only included in a single place, so let's refactor this when we need it.